Delete containers/terracotta-server directory

mathieucarbou opened the pull request:
The terracotta information that is there is outdated, contains many issues, and cannot be supported by terracotta R&D. It SHOULD NOT BE USED IN ANY CASE.

- BigMemory official docker images and doc: https://containers.softwareag.com/products/bigmemorymax-server

- Terracotta DB official docker images and doc: https://containers.softwareag.com/products/terracotta-server

- Helm charts and other doc and links: https://github.softwareag.com/Terracotta/

The locations above are official Docker images deployed and supported by Terracotta R&D.

Ref: https://itrac.eur.ad.sag/browse/TAB-8504

YanaSimeonova reviewed (commented):
Hi TC team, let me check with doc team how to handle this. We have deprecated the CCE docker builder, and there was a plan to remove it all. But if the template is no longer supported and not working it will be handled in the readme and not removing it directly because it is used in older versions too.

YanaSimeonova commented:
@mathieucarbou, we have deleted docker builder parts. Can we close this pull request?

mathieucarbou commented:
> @mathieucarbou, we have deleted docker builder parts. Can we close this pull request?

No, the templates were not deleted. I have rebased this PR, please see the diff.
